src/Hosting/TestHost/src/ResponseStream.cs

```diff
@@ -116,6 +117,7 @@ public async override Task<int> ReadAsync(byte[] buffer, int offset, int count,
                 if (result.Buffer.IsEmpty && result.IsCompleted)
                 {
                     _pipe.Reader.Complete();
+                    _readComplete();
```

Comment:
ReadAsync can be called multiple times at the end and return 0 each time, no? Or would the pipe reader throw if you did that? You don't want _readComplete being called multiple times.

Reply:
The pipe reader throws. I've updated the stream to return 0 when the reader is complete.

Can you double check the way I've done it?